### PR TITLE
Allow to add extra params to callback_url

### PR DESCRIPTION
**Pull request by radzserg**
I added ability to specify :callback_params in settings for strategies 

 
```
config :ueberauth, Ueberauth,
   providers: [
     facebook: { Ueberauth.Strategy.Facebook, [callback_params: ["type", "user_role"]]} 
]
```

then add this extra param to login/sign-up links 
```

 auth_path(@conn, :request, "facebook", type: "sign-in")
 auth_path(@conn, :request, "twitter", user_role: "manager")
```

Tested on facebook twitter and VK strategies.

### CODE REVIEW

File: test/ueberauth_test.exs [hunk 1]
```diff
@@ -118,15 +119,18 @@ defmodule UeberauthTest do
   test "callback_url port" do
     conn = %{conn(:get, "/") | scheme: :https, port: 80}
     conn = put_private(conn, :ueberauth_request_options, [callback_path: "/auth/provider/callback"])
+    conn =  %{conn | params: %{}}
+
     assert Ueberauth.Strategy.Helpers.callback_url(conn) ==
       "https://www.example.com/auth/provider/callback"
   end
 
-  test "callback_url forwarded protocol" do
```
Comment (doomspork) [inline test/ueberauth_test.exs]: Do we want to remove this test?

File: test/ueberauth_test.exs [hunk 2]
```diff
@@ -14,7 +14,8 @@ defmodule UeberauthTest do
   end
 
   test "simple callback phase" do
-    conn = conn(:get, "/auth/simple/callback") |> SpecRouter.call(@opts)
+    conn = conn(:get, "/auth/simple/callback")
```
Comment (doomspork) [inline test/ueberauth_test.exs]: Would you mind refactoring this bit?   I need to setup Credo on the repos.

```elixir
conn = :get
       |> conn("/auth/simple/callback")
       |> SpecRouter.call(@opts)
auth = conn.assigns.ueberauth_auth
```

**Comment by radzserg** [comment]
@doomspork  Done

**Comment by doomspork** [comment]
Thank you @radzserg 👍

**Pull request merged**